src/main/java/com/shaneschulte/mc/clanarena/commands/CreateCmd.java

```diff
@@ -0,0 +1,379 @@
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class CreateCmd implements CmdProperties, Listener {
+
+    private static HashMap<String, BossBar> currentPlayerBossBars = new HashMap<>();
+    private static HashMap<String, String> currentArenaNameByPlayer = new HashMap<>();
+    private static int minSpawns = 6;
+
+    // Stage 01: selecting region
+    private static ArrayList<String> playersInTheMiddleOfDesignatingThe2PointsNeededToCreateAnArenaRegion = new ArrayList<>();
+
+    private static HashMap<String, Location> currentPos1ByPlayer = new HashMap<>();
+    private static HashMap<String, Location> currentPos2ByPlayer = new HashMap<>();
+
+    // Stage 02: selecting spawn points
+    private static ArrayList<String> playersInTheMiddleOfSelectingSpawnPoints = new ArrayList<>();
+    private static ArrayList<String> playersAddingSecondTeamSpawnPoints = new ArrayList<>();
+
+    private static HashMap<String, ArrayList> currentTeam1SpawnsByPlayer = new HashMap<>();
+    private static HashMap<String, ArrayList> currentTeam2SpawnsByPlayer = new HashMap<>();
+
+    @Override
+    public void perform (Player p, String allArgs, String[] args) {
+        // if just /ca create
+        if (args.length <= 1) {
+            MsgUtils.error(p, "please specify a " + MsgUtils.Colors.VARIABLE + "name");
+        }
+
+        else if (isPlayerAlreadyCreating(p.getName())) {
+            MsgUtils.error(p, "You are already in the middle of creating an arena. Scroll up in chat if you need help or disconnect // die to reset this process");
+        }
+
+        // if /ca create <name>
+        else {
+            String name = args[1];
+            currentArenaNameByPlayer.put(p.getName(), args[1]);
+            playersInTheMiddleOfDesignatingThe2PointsNeededToCreateAnArenaRegion.add(p.getName());
+            MsgUtils.success(p,"Creating arena \"" + MsgUtils.Colors.VARIABLE + name + MsgUtils.Colors.SUCCESS + "\"");
+            MsgUtils.sendMessage(p, MsgUtils.Colors.VARIABLE + "Left Click " + MsgUtils.Colors.INFO + "the first point in the " + MsgUtils.Colors.VARIABLE + "Arena" + MsgUtils.Colors.INFO + "'s region");
+            MsgUtils.sendMessage(p, MsgUtils.Colors.VARIABLE + "Right Click " + MsgUtils.Colors.INFO + "the second point in the " + MsgUtils.Colors.VARIABLE + "Arena" + MsgUtils.Colors.INFO + "'s region");
+
+            BossBar bar = Bukkit.getServer().createBossBar("PROGRESS", BarColor.PINK, BarStyle.SEGMENTED_12);
+            currentPlayerBossBars.put(p.getName(), bar);
+            bar.addPlayer(p);
+            updateProgressBossBar(p.getName());
+        }
+
+    }
+
+    /**
+     * Used to set the region in stage 01
+     */
+    @EventHandler
+    public void onPlayerInteract (PlayerInteractEvent event) {
```

Comment:
I'm not in love with the idea of having this class implement Listener, but that can potentially be refactored later